providers/openshift/provider.go

```diff
@@ -349,7 +351,10 @@ func (p *OpenShiftProvider) ValidateRequest(req *http.Request) (*providers.Sessi
 		log.Printf("authorizer reason: %s", reason)
 		return nil, nil
 	}
-	return &providers.SessionState{User: user.GetName(), Email: user.GetName() + "@cluster.local"}, nil
+
+	parts := strings.SplitN(auth, " ", 2)
+	token := parts[1]
```

Comment:
This code will forward any authorization, for example Basic Auth iwth username/password performede by a user.
Is this intentional ?
If so it requires an explicit warning in the documentation.
If not the authorization type (parts[0]) should probably be matched first.

Reply:
It seems that we would only get here with a basic auth header if the password is incorrect, and then the attempt to authorize a bearer token with the basic auth header would fail, so we would never actually forward it. However, I would prefer there to be a check against basic anyways for future safety.